Fix a lot of errors for updating to 1.13

Ultra03 opened the pull request:
I updated a lot of the Materials to their 1.13 versions, which fixed most of the 1.13 incompatibilities (not including deprecations). I used the legacy versions where I wasn't sure how using a new material would affect it. There are some errors I'm not quite sure how to fix, such as undefined ItemStack constructors.

Edit: It might be best to put this into a new branch, to separate the older versions with 1.13.

Ultra03 committed:
Fix a lot of errors in updating to 1.13

cmastudios commented:
Nice, I'll try to test it later this week maybe.

IMegaLightI commented:
A new update for 1.13 would be cool, it's vital for us but take the time. :)

Thephoenix844 commented:
Thanks you very much !

cmastudios commented:
This does not compile on my machine:

```
[INFO] --- maven-compiler-plugin:2.0.2:compile (default-compile) @ war ---
[INFO] Compiling 140 source files to /home/connor/Documents/Minecraft/war/war/target/classes
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  2.244 s
[INFO] Finished at: 2018-11-12T13:09:32-06:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:2.0.2:compile (default-compile) on project war: Compilation failure: Compilation failure: 
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/LoadoutTxtMapper.java:[21,42] error: incompatible types: int cannot be converted to Material
[ERROR] 
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/LoadoutTxtMapper.java:[24,42] error: incompatible types: int cannot be converted to Material
[ERROR] 
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/LoadoutTxtMapper.java:[28,42] error: incompatible types: int cannot be converted to Material
[ERROR] 
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/LoadoutTxtMapper.java:[38,46] error: cannot find symbol
[ERROR] 
[ERROR] could not parse error message:   symbol:   method getById(int)
[ERROR]   location: class Enchantment
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/LoadoutYmlMapper.java:66: error: incompatible types: int cannot be converted to Material
[ERROR] 			ItemStack stack = new ItemStack(id, amount, durability);
[ERROR] 			                                ^
[ERROR] 
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/LoadoutYmlMapper.java:[76,46] error: cannot find symbol
[ERROR] 
[ERROR] could not parse error message:   symbol:   method getById(int)
[ERROR]   location: class Enchantment
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/PreNimitzZoneVolumeMapper.java:123: error: cannot find symbol
[ERROR] 									worldBlockId = worldBlock.getTypeId();
[ERROR] 									                         ^
[ERROR] 
[ERROR] could not parse error message:   symbol:   method getTypeId()
[ERROR]   location: variable worldBlock of type Block
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/PreNimitzZoneVolumeMapper.java:124: error: cannot find symbol
[ERROR] 									if (worldBlockId != diskBlockType || (worldBlockId == diskBlockType && worldBlock.getData() != diskBlockData) || (worldBlockId == diskBlockType && worldBlock.getData() == diskBlockData && (diskBlockType == Material.WALL_SIGN.getId() || diskBlockType == Material.SIGN_POST.getId() || diskBlockType == Material.CHEST.getId() || diskBlockType == Material.DISPENSER.getId()))) {
[ERROR] 									                                                                                                                                                                                                                                                                     ^
[ERROR] 
[ERROR] could not parse error message:   symbol:   variable SIGN_POST
[ERROR]   location: class Material
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/PreNimitzZoneVolumeMapper.java:125: error: cannot find symbol
[ERROR] 										if (diskBlockType == Material.WALL_SIGN.getId() || diskBlockType == Material.SIGN_POST.getId()) {
[ERROR] 										                                                                            ^
[ERROR] 
[ERROR] could not parse error message:   symbol:   variable SIGN_POST
[ERROR]   location: class Material
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/PreNimitzZoneVolumeMapper.java:135: error: incompatible types: int cannot be converted to String
[ERROR] 												worldBlock.setType(Material.getMaterial(diskBlockType));
[ERROR] 												                                        ^
[ERROR] 
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/PreNimitzZoneVolumeMapper.java:[137,55] error: incompatible types: int cannot be converted to Material
[ERROR] 
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/PreNimitzZoneVolumeMapper.java:[162,51] error: incompatible types: int cannot be converted to String
[ERROR] 
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/PreNimitzZoneVolumeMapper.java:[163,21] error: cannot find symbol
[ERROR] 
[ERROR] could not parse error message:   symbol:   method setData(byte)
[ERROR]   location: variable worldBlock of type Block
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/PreNimitzZoneVolumeMapper.java:184: error: incompatible types: int cannot be converted to String
[ERROR] 											worldBlock.setType(Material.getMaterial(diskBlockType));
[ERROR] 											                                        ^
[ERROR] 
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/PreNimitzZoneVolumeMapper.java:[185,21] error: cannot find symbol
[ERROR] 
[ERROR] could not parse error message:   symbol:   method setData(byte)
[ERROR]   location: variable worldBlock of type Block
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/PreNimitzZoneVolumeMapper.java:201: error: cannot find symbol
[ERROR] 										} else if (diskBlockType == Material.WOODEN_DOOR.getId() || diskBlockType == Material.IRON_DOOR_BLOCK.getId()) {
[ERROR] 										                                    ^
[ERROR] 
[ERROR] could not parse error message:   symbol:   variable WOODEN_DOOR
[ERROR]   location: class Material
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/PreNimitzZoneVolumeMapper.java:201: error: cannot find symbol
[ERROR] 										} else if (diskBlockType == Material.WOODEN_DOOR.getId() || diskBlockType == Material.IRON_DOOR_BLOCK.getId()) {
[ERROR] 										                                                                                     ^
[ERROR] 
[ERROR] could not parse error message:   symbol:   variable IRON_DOOR_BLOCK
[ERROR]   location: class Material
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/PreNimitzZoneVolumeMapper.java:204: error: cannot find symbol
[ERROR] 										} else if (((diskBlockType == Material.TORCH.getId() && ((diskBlockData & 0x02) == 0x02)) || (diskBlockType == Material.REDSTONE_TORCH_OFF.getId() && ((diskBlockData & 0x02) == 0x02)) || (diskBlockType == Material.REDSTONE_TORCH_ON.getId() && ((diskBlockData & 0x02) == 0x02)) || (diskBlockType == Material.LEVER.getId() && ((diskBlockData & 0x02) == 0x02)) || (diskBlockType == Material.STONE_BUTTON.getId() && ((diskBlockData & 0x02) == 0x02)) || (diskBlockType == Material.LADDER.getId() && ((diskBlockData & 0x04) == 0x04)) || (diskBlockType == Material.RAILS.getId() && ((diskBlockData & 0x02) == 0x02))) && i + 1 != volume.getSizeX()) {
[ERROR] 										                                                                                                                       ^
[ERROR] 
[ERROR] could not parse error message:   symbol:   variable REDSTONE_TORCH_OFF
[ERROR]   location: class Material
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/PreNimitzZoneVolumeMapper.java:204: error: cannot find symbol
[ERROR] 										} else if (((diskBlockType == Material.TORCH.getId() && ((diskBlockData & 0x02) == 0x02)) || (diskBlockType == Material.REDSTONE_TORCH_OFF.getId() && ((diskBlockData & 0x02) == 0x02)) || (diskBlockType == Material.REDSTONE_TORCH_ON.getId() && ((diskBlockData & 0x02) == 0x02)) || (diskBlockType == Material.LEVER.getId() && ((diskBlockData & 0x02) == 0x02)) || (diskBlockType == Material.STONE_BUTTON.getId() && ((diskBlockData & 0x02) == 0x02)) || (diskBlockType == Material.LADDER.getId() && ((diskBlockData & 0x04) == 0x04)) || (diskBlockType == Material.RAILS.getId() && ((diskBlockData & 0x02) == 0x02))) && i + 1 != volume.getSizeX()) {
[ERROR] 										                                                                                                                                                                                                                     ^
[ERROR] 
[ERROR] could not parse error message:   symbol:   variable REDSTONE_TORCH_ON
[ERROR]   location: class Material
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/PreNimitzZoneVolumeMapper.java:204: error: cannot find symbol
[ERROR] 										} else if (((diskBlockType == Material.TORCH.getId() && ((diskBlockData & 0x02) == 0x02)) || (diskBlockType == Material.REDSTONE_TORCH_OFF.getId() && ((diskBlockData & 0x02) == 0x02)) || (diskBlockType == Material.REDSTONE_TORCH_ON.getId() && ((diskBlockData & 0x02) == 0x02)) || (diskBlockType == Material.LEVER.getId() && ((diskBlockData & 0x02) == 0x02)) || (diskBlockType == Material.STONE_BUTTON.getId() && ((diskBlockData & 0x02) == 0x02)) || (diskBlockType == Material.LADDER.getId() && ((diskBlockData & 0x04) == 0x04)) || (diskBlockType == Material.RAILS.getId() && ((diskBlockData & 0x02) == 0x02))) && i + 1 != volume.getSizeX()) {
[ERROR] 										                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             ^
[ERROR] 
[ERROR] could not parse error message:   symbol:   variable RAILS
[ERROR]   location: class Material
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/PreNimitzZoneVolumeMapper.java:211: error: incompatible types: int cannot be converted to String
[ERROR] 												worldBlock.setType(Material.getMaterial(diskBlockType));
[ERROR] 												                                        ^
[ERROR] 
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/PreNimitzZoneVolumeMapper.java:[212,22] error: cannot find symbol
[ERROR] 
[ERROR] could not parse error message:   symbol:   method setData(byte)
[ERROR]   location: variable worldBlock of type Block
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/PreNimitzZoneVolumeMapper.java:219: error: incompatible types: int cannot be converted to String
[ERROR] 												worldBlock.setType(Material.getMaterial(256 + diskBlockType));
[ERROR] 												                                            ^
[ERROR] 
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/PreNimitzZoneVolumeMapper.java:[220,22] error: cannot find symbol
[ERROR] 
[ERROR] could not parse error message:   symbol:   method setData(byte)
[ERROR]   location: variable worldBlock of type Block
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/PreNimitzZoneVolumeMapper.java:334: error: cannot find symbol
[ERROR] 								typeId = block.getTypeId();
[ERROR] 								              ^
[ERROR] 
[ERROR] could not parse error message:   symbol:   method getTypeId()
[ERROR]   location: variable block of type Block
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/PreDeGaulleZoneVolumeMapper.java:41: error: incompatible types: int cannot be converted to Material
[ERROR] 				ItemStack stack = new ItemStack(Integer.parseInt(itemStrSplit[0]), Integer.parseInt(itemStrSplit[1]));
[ERROR] 				                                                ^
[ERROR] 
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/PreDeGaulleZoneVolumeMapper.java:[42,40] error: cannot find symbol
[ERROR] 
[ERROR] could not parse error message:   symbol:   method getTypeId()
[ERROR]   location: variable stack of type ItemStack
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/PreDeGaulleZoneVolumeMapper.java:47: error: incompatible types: int cannot be converted to Material
[ERROR] 				ItemStack stack = new ItemStack(Integer.parseInt(itemStrSplit[0]), Integer.parseInt(itemStrSplit[1]));
[ERROR] 				                                                ^
[ERROR] 
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/PreDeGaulleZoneVolumeMapper.java:[52,44] error: incompatible types: int cannot be converted to Material
[ERROR] 
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/PreDeGaulleZoneVolumeMapper.java:[111,36] error: cannot find symbol
[ERROR] 
[ERROR] could not parse error message:   symbol:   method getTypeId()
[ERROR]   location: variable worldBlock of type Block
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/PreDeGaulleZoneVolumeMapper.java:112: error: cannot find symbol
[ERROR] 											if (worldBlockId != diskBlockType || (worldBlockId == diskBlockType && worldBlock.getData() != diskBlockData) || (worldBlockId == diskBlockType && worldBlock.getData() == diskBlockData && (diskBlockType == Material.WALL_SIGN.getId() || diskBlockType == Material.SIGN_POST.getId() || diskBlockType == Material.CHEST.getId() || diskBlockType == Material.DISPENSER.getId()))) {
[ERROR] 											                                                                                                                                                                                                                                                                     ^
[ERROR] 
[ERROR] could not parse error message:   symbol:   variable SIGN_POST
[ERROR]   location: class Material
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/PreDeGaulleZoneVolumeMapper.java:113: error: cannot find symbol
[ERROR] 												if (diskBlockType == Material.WALL_SIGN.getId() || diskBlockType == Material.SIGN_POST.getId()) {
[ERROR] 												                                                                            ^
[ERROR] 
[ERROR] could not parse error message:   symbol:   variable SIGN_POST
[ERROR]   location: class Material
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/PreDeGaulleZoneVolumeMapper.java:124: error: cannot find symbol
[ERROR] 														if (diskBlockType == Material.SIGN_POST.getId() && ((diskBlockData & 0x04) == 0x04) && i + 1 != volume.getSizeX()) {
[ERROR] 														                             ^
[ERROR] 
[ERROR] could not parse error message:   symbol:   variable SIGN_POST
[ERROR]   location: class Material
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/PreDeGaulleZoneVolumeMapper.java:127: error: incompatible types: int cannot be converted to String
[ERROR] 															worldBlock.setType(Material.getMaterial(diskBlockType));
[ERROR] 															                                        ^
[ERROR] 
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/PreDeGaulleZoneVolumeMapper.java:[129,58] error: incompatible types: int cannot be converted to Material
[ERROR] 
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/PreDeGaulleZoneVolumeMapper.java:[159,53] error: incompatible types: int cannot be converted to String
[ERROR] 
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/PreDeGaulleZoneVolumeMapper.java:[160,23] error: cannot find symbol
[ERROR] 
[ERROR] could not parse error message:   symbol:   method setData(byte)
[ERROR]   location: variable worldBlock of type Block
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/PreDeGaulleZoneVolumeMapper.java:186: error: incompatible types: int cannot be converted to String
[ERROR] 													worldBlock.setType(Material.getMaterial(diskBlockType));
[ERROR] 													                                        ^
[ERROR] 
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/PreDeGaulleZoneVolumeMapper.java:[187,23] error: cannot find symbol
[ERROR] 
[ERROR] could not parse error message:   symbol:   method setData(byte)
[ERROR]   location: variable worldBlock of type Block
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/PreDeGaulleZoneVolumeMapper.java:203: error: cannot find symbol
[ERROR] 												} else if (diskBlockType == Material.WOODEN_DOOR.getId() || diskBlockType == Material.IRON_DOOR_BLOCK.getId()) {
[ERROR] 												                                    ^
[ERROR] 
[ERROR] could not parse error message:   symbol:   variable WOODEN_DOOR
[ERROR]   location: class Material
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/PreDeGaulleZoneVolumeMapper.java:203: error: cannot find symbol
[ERROR] 												} else if (diskBlockType == Material.WOODEN_DOOR.getId() || diskBlockType == Material.IRON_DOOR_BLOCK.getId()) {
[ERROR] 												                                                                                     ^
[ERROR] 
[ERROR] could not parse error message:   symbol:   variable IRON_DOOR_BLOCK
[ERROR]   location: class Material
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/PreDeGaulleZoneVolumeMapper.java:208: error: cannot find symbol
[ERROR] 														boolean belowIsGlass = blockBelow.getTypeId() == Material.GLASS.getId();
[ERROR] 														                                 ^
[ERROR] 
[ERROR] could not parse error message:   symbol:   method getTypeId()
[ERROR]   location: variable blockBelow of type Block
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/PreDeGaulleZoneVolumeMapper.java:213: error: incompatible types: int cannot be converted to String
[ERROR] 															blockBelow.setType(Material.getMaterial(diskBlockType));
[ERROR] 															                                        ^
[ERROR] 
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/PreDeGaulleZoneVolumeMapper.java:[214,25] error: cannot find symbol
[ERROR] 
[ERROR] could not parse error message:   symbol:   method setData(byte)
[ERROR]   location: variable blockBelow of type Block
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/PreDeGaulleZoneVolumeMapper.java:215: error: incompatible types: int cannot be converted to String
[ERROR] 															worldBlock.setType(Material.getMaterial(diskBlockType));
[ERROR] 															                                        ^
[ERROR] 
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/PreDeGaulleZoneVolumeMapper.java:[216,25] error: cannot find symbol
[ERROR] 
[ERROR] could not parse error message:   symbol:   method setData(byte)
[ERROR]   location: variable worldBlock of type Block
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/PreDeGaulleZoneVolumeMapper.java:222: error: cannot find symbol
[ERROR] 												} else if (((diskBlockType == Material.TORCH.getId() && ((diskBlockData & 0x02) == 0x02)) || (diskBlockType == Material.REDSTONE_TORCH_OFF.getId() && ((diskBlockData & 0x02) == 0x02)) || (diskBlockType == Material.REDSTONE_TORCH_ON.getId() && ((diskBlockData & 0x02) == 0x02)) || (diskBlockType == Material.LEVER.getId() && ((diskBlockData & 0x02) == 0x02)) || (diskBlockType == Material.STONE_BUTTON.getId() && ((diskBlockData & 0x02) == 0x02)) || (diskBlockType == Material.LADDER.getId() && ((diskBlockData & 0x04) == 0x04)) || (diskBlockType == Material.RAILS.getId() && ((diskBlockData & 0x02) == 0x02))) && i + 1 != volume.getSizeX()) {
[ERROR] 												                                                                                                                       ^
[ERROR] 
[ERROR] could not parse error message:   symbol:   variable REDSTONE_TORCH_OFF
[ERROR]   location: class Material
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/PreDeGaulleZoneVolumeMapper.java:222: error: cannot find symbol
[ERROR] 												} else if (((diskBlockType == Material.TORCH.getId() && ((diskBlockData & 0x02) == 0x02)) || (diskBlockType == Material.REDSTONE_TORCH_OFF.getId() && ((diskBlockData & 0x02) == 0x02)) || (diskBlockType == Material.REDSTONE_TORCH_ON.getId() && ((diskBlockData & 0x02) == 0x02)) || (diskBlockType == Material.LEVER.getId() && ((diskBlockData & 0x02) == 0x02)) || (diskBlockType == Material.STONE_BUTTON.getId() && ((diskBlockData & 0x02) == 0x02)) || (diskBlockType == Material.LADDER.getId() && ((diskBlockData & 0x04) == 0x04)) || (diskBlockType == Material.RAILS.getId() && ((diskBlockData & 0x02) == 0x02))) && i + 1 != volume.getSizeX()) {
[ERROR] 												                                                                                                                                                                                                                     ^
[ERROR] 
[ERROR] could not parse error message:   symbol:   variable REDSTONE_TORCH_ON
[ERROR]   location: class Material
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/PreDeGaulleZoneVolumeMapper.java:222: error: cannot find symbol
[ERROR] 												} else if (((diskBlockType == Material.TORCH.getId() && ((diskBlockData & 0x02) == 0x02)) || (diskBlockType == Material.REDSTONE_TORCH_OFF.getId() && ((diskBlockData & 0x02) == 0x02)) || (diskBlockType == Material.REDSTONE_TORCH_ON.getId() && ((diskBlockData & 0x02) == 0x02)) || (diskBlockType == Material.LEVER.getId() && ((diskBlockData & 0x02) == 0x02)) || (diskBlockType == Material.STONE_BUTTON.getId() && ((diskBlockData & 0x02) == 0x02)) || (diskBlockType == Material.LADDER.getId() && ((diskBlockData & 0x04) == 0x04)) || (diskBlockType == Material.RAILS.getId() && ((diskBlockData & 0x02) == 0x02))) && i + 1 != volume.getSizeX()) {
[ERROR] 												                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             ^
[ERROR] 
[ERROR] could not parse error message:   symbol:   variable RAILS
[ERROR]   location: class Material
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/PreDeGaulleZoneVolumeMapper.java:228: error: incompatible types: int cannot be converted to String
[ERROR] 													worldBlock.setType(Material.getMaterial(diskBlockType));
[ERROR] 													                                        ^
[ERROR] 
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/PreDeGaulleZoneVolumeMapper.java:[229,23] error: cannot find symbol
[ERROR] 
[ERROR] could not parse error message:   symbol:   method setData(byte)
[ERROR]   location: variable worldBlock of type Block
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/WarzoneYmlMapper.java:393: error: incompatible types: int cannot be converted to Material
[ERROR] 						new ItemStack(floorMaterialSection.getInt("id"), 1,
[ERROR] 						                                         ^
[ERROR] 
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/WarzoneYmlMapper.java:[405,47] error: incompatible types: int cannot be converted to Material
[ERROR] 
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/WarzoneYmlMapper.java:[417,47] error: incompatible types: int cannot be converted to Material
[ERROR] 
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/WarzoneYmlMapper.java:[429,47] error: incompatible types: int cannot be converted to Material
[ERROR] 
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/WarzoneYmlMapper.java:[457,47] error: incompatible types: int cannot be converted to Material
[ERROR] 
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/WarzoneYmlMapper.java:[469,47] error: incompatible types: int cannot be converted to Material
[ERROR] 
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/WarzoneYmlMapper.java:[481,47] error: incompatible types: int cannot be converted to Material
[ERROR] 
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/job/RestoreYmlWarhubJob.java:[43,46] error: incompatible types: int cannot be converted to Material
[ERROR] 
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/job/RestoreYmlWarhubJob.java:[55,46] error: incompatible types: int cannot be converted to Material
[ERROR] 
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/job/RestoreYmlWarhubJob.java:[67,46] error: incompatible types: int cannot be converted to Material
[ERROR] 
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/job/RestoreYmlWarhubJob.java:[79,46] error: incompatible types: int cannot be converted to Material
[ERROR] 
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/ZoneVolumeMapper.java:[317,8] error: an enum switch case label must be the unqualified name of an enumeration constant
[ERROR] 
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/ZoneVolumeMapper.java:[338,8] error: an enum switch case label must be the unqualified name of an enumeration constant
[ERROR] 
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/VolumeMapper.java:[140,15] error: cannot find symbol
[ERROR] 
[ERROR] could not parse error message:   symbol:   method setTypeId(int)
[ERROR]   location: variable dummy of type BlockState
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/VolumeMapper.java:244: error: incompatible types: int cannot be converted to Material
[ERROR] 					ItemStack stack = new ItemStack(Integer.parseInt(itemStrSplit[0]), Integer.parseInt(itemStrSplit[1]));
[ERROR] 					                                                ^
[ERROR] 
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/VolumeMapper.java:[245,41] error: cannot find symbol
[ERROR] 
[ERROR] could not parse error message:   symbol:   method getTypeId()
[ERROR]   location: variable stack of type ItemStack
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/VolumeMapper.java:256: error: cannot find symbol
[ERROR] 							War.war.safelyEnchant(stack, Enchantment.getById(enchantId), level);
[ERROR] 							                                        ^
[ERROR] 
[ERROR] could not parse error message:   symbol:   method getById(int)
[ERROR]   location: class Enchantment
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/VolumeMapper.java:262: error: incompatible types: int cannot be converted to Material
[ERROR] 					ItemStack stack = new ItemStack(Integer.parseInt(itemStrSplit[0]), Integer.parseInt(itemStrSplit[1]));
[ERROR] 					                                                ^
[ERROR] 
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/VolumeMapper.java:[263,41] error: cannot find symbol
[ERROR] 
[ERROR] could not parse error message:   symbol:   method getTypeId()
[ERROR]   location: variable stack of type ItemStack
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/VolumeMapper.java:268: error: incompatible types: int cannot be converted to Material
[ERROR] 					ItemStack stack = new ItemStack(Integer.parseInt(itemStrSplit[0]), Integer.parseInt(itemStrSplit[1]));
[ERROR] 					                                                ^
[ERROR] 
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/VolumeMapper.java:[273,45] error: incompatible types: int cannot be converted to Material
[ERROR] 
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/VolumeMapper.java:[291,17] error: cannot find symbol
[ERROR] 
[ERROR] could not parse error message:   symbol:   method getTypeId()
[ERROR]   location: variable item of type ItemStack
[ERROR] /home/connor/Documents/Minecraft/war/war/src/main/java/com/tommytony/war/mapper/VolumeMapper.java:298: error: cannot find symbol
[ERROR] 						enchantmentsStr += enchantment.getId() + ":" + item.getEnchantments().get(enchantment) + "::";
[ERROR] 						                              ^

```

IMegaLightI commented:
Ohh yes there is a lot of error :/

Ultra03 commented:
@cmastudios Yeah, there are still some errors I’m not sure how to fix (new ItemStacks, etc.) :/

I can possibly try to fix these sometime soon, but I’m not quite sure how atm.

IMegaLightI commented:
Of course 1.13 add a lot of new thing for the block, unfortunately a lot of plugins need to change a part of their code

cmastudios commented:
@Ultra03 alright I may have time this Sunday to work on those issues.

Thephoenix844 commented:
So you have succeeded? 😁

cmastudios committed:
Fix renamed paintings

cmastudios committed:
Drop all War backwards compatibility

Bukkit 1.13 removes item ID API entirely - therefore, it is impossible
to load legacy War format files that still use item IDs.

cmastudios committed:
Fix some deprecated stuff

cmastudios committed:
Bump War major version due to breaking changes

cmastudios commented:
Fixed all of the compile-time errors. Could any/all interested parties help test this new build of War for Minecraft 1.13?

War Build: https://s3-us-west-2.amazonaws.com/cmastore/minecraft/war-2.0-SNAPSHOT.jar

Spigot Build: https://s3-us-west-2.amazonaws.com/cmastore/minecraft/spigot-1.13.2.jar

(Make backups before updating)

cmastudios committed:
Fix SQL bug

cmastudios committed:
Fix wrong color for team gate

cmastudios committed:
Fix UI glitches with WarHub and Dye color updates

cmastudios commented:
Build updated: https://s3-us-west-2.amazonaws.com/cmastore/minecraft/war-2.0-SNAPSHOT.jar

IMegaLightI commented:
There is an error look I send it to you if you want to see @cmastudios

[12:20:53 INFO]: MegaLight issued server command: /war
[12:21:07 WARN]: [War] Task #8233 for War v2.0-SNAPSHOT generated an exception
java.lang.NoSuchMethodError: com.sk89q.worldedit.bukkit.WorldEditPlugin.getSelection(Lorg/bukkit/entity/Player;)Lcom/sk89q/worldedit/bukkit/selections/Selection;
at com.tommytony.war.ui.EditOrCreateZoneUI$1$1.run(EditOrCreateZoneUI.java:42) ~[?:?]
at org.bukkit.craftbukkit.v1_13_R2.scheduler.CraftTask.run(CraftTask.java:76) ~[minecraft_server.jar:git-Spigot-2474d93-f0398e4]
at org.bukkit.craftbukkit.v1_13_R2.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:361) [minecraft_server.jar:git-Spigot-2474d93-f0398e4]
at net.minecraft.server.v1_13_R2.MinecraftServer.b(MinecraftServer.java:893) [minecraft_server.jar:git-Spigot-2474d93-f0398e4]
at net.minecraft.server.v1_13_R2.DedicatedServer.b(DedicatedServer.java:417) [minecraft_server.jar:git-Spigot-2474d93-f0398e4]
at net.minecraft.server.v1_13_R2.MinecraftServer.a(MinecraftServer.java:835) [minecraft_server.jar:git-Spigot-2474d93-f0398e4]
at net.minecraft.server.v1_13_R2.MinecraftServer.run(MinecraftServer.java:733) [minecraft_server.jar:git-Spigot-2474d93-f0398e4]
at java.lang.Thread.run(Thread.java:748) [?:1.8.0_181]

cmastudios commented:
I suppose this is right after you tried to create a zone?

IMegaLightI commented:
yep

IMegaLightI commented:
after i tired to enter the name of the zone

cmastudios commented:
Hm ok, does the /setzone command work fine?

IMegaLightI commented:
nop i got that [16:54:37 INFO]: MegaLight issued server command: /setzone test
[16:54:37 ERROR]: null
org.bukkit.command.CommandException: Unhandled exception executing command 'setzone' in plugin War v2.0-SNAPSHOT
at org.bukkit.command.PluginCommand.execute(PluginCommand.java:46) ~[minecraft_server.jar:git-Spigot-2474d93-f0398e4]
at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:139) ~[minecraft_server.jar:git-Spigot-2474d93-f0398e4]
at org.bukkit.craftbukkit.v1_13_R2.CraftServer.dispatchCommand(CraftServer.java:699) ~[minecraft_server.jar:git-Spigot-2474d93-f0398e4]
at net.minecraft.server.v1_13_R2.PlayerConnection.handleCommand(PlayerConnection.java:1648) ~[minecraft_server.jar:git-Spigot-2474d93-f0398e4]
at net.minecraft.server.v1_13_R2.PlayerConnection.a(PlayerConnection.java:1483) ~[minecraft_server.jar:git-Spigot-2474d93-f0398e4]
at net.minecraft.server.v1_13_R2.PacketPlayInChat.a(PacketPlayInChat.java:45) ~[minecraft_server.jar:git-Spigot-2474d93-f0398e4]
at net.minecraft.server.v1_13_R2.PacketPlayInChat.a(PacketPlayInChat.java:1) ~[minecraft_server.jar:git-Spigot-2474d93-f0398e4]
at net.minecraft.server.v1_13_R2.PlayerConnectionUtils.a(SourceFile:10) ~[minecraft_server.jar:git-Spigot-2474d93-f0398e4]
at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_181]
at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_181]
at net.minecraft.server.v1_13_R2.SystemUtils.a(SourceFile:199) [minecraft_server.jar:git-Spigot-2474d93-f0398e4]
at net.minecraft.server.v1_13_R2.MinecraftServer.b(MinecraftServer.java:900) [minecraft_server.jar:git-Spigot-2474d93-f0398e4]
at net.minecraft.server.v1_13_R2.DedicatedServer.b(DedicatedServer.java:417) [minecraft_server.jar:git-Spigot-2474d93-f0398e4]
at net.minecraft.server.v1_13_R2.MinecraftServer.a(MinecraftServer.java:835) [minecraft_server.jar:git-Spigot-2474d93-f0398e4]
at net.minecraft.server.v1_13_R2.MinecraftServer.run(MinecraftServer.java:733) [minecraft_server.jar:git-Spigot-2474d93-f0398e4]
at java.lang.Thread.run(Thread.java:748) [?:1.8.0_181]
Caused by: java.lang.NoSuchMethodError: com.sk89q.worldedit.bukkit.WorldEditPlugin.getSelection(Lorg/bukkit/entity/Player;)Lcom/sk89q/worldedit/bukkit/selections/Selection;
at com.tommytony.war.command.SetZoneCommand.handle(SetZoneCommand.java:35) ~[?:?]
at com.tommytony.war.command.WarCommandHandler.handle(WarCommandHandler.java:141) ~[?:?]
at com.tommytony.war.War.onCommand(War.java:344) ~[?:?]
at org.bukkit.command.PluginCommand.execute(PluginCommand.java:44) ~[minecraft_server.jar:git-Spigot-2474d93-f0398e4]
... 15 more

cmastudios commented:
I'll look into it. What version of WorldEdit are you running? If you disable WorldEdit for now you should be able to do further testing.

IMegaLightI commented:
worldedit-bukkit-7.0.0-beta-01

i can tried to disable worldedit if you want

cmastudios commented:
Yeah disable it for now and you can do further testing.

IMegaLightI commented:
ok its disable

IMegaLightI commented:
what can i do for you ?

cmastudios commented:
Can you verify that the plugin functions correctly? Try playing in a zone, changing settings, saving and resetting, restarting the server, etc.

IMegaLightI commented:
ok but without worldedit i can't create zone so i can't test

cmastudios commented:
You can create warzones without WorldEdit, after removing WorldEdit use the /setzone command.

IMegaLightI commented:
ok

IMegaLightI commented:
Yea without worldedit the plugin works perfectly

cmastudios committed:
Update WorldEdit Integration

Breaking, now requires WorldEdit newer than Oct. 18, 2018

cmastudios commented:
Updated version with WorldEdit fix: https://s3-us-west-2.amazonaws.com/maven.cmastudios.me/com/tommytony/war/2.0-SNAPSHOT/war-2.0-20181119.195922-2.jar
Requires WorldEdit newer than Oct. 18, 2018

IMegaLightI commented:
I'm sorry to tell you that it still does not work

[21:07:48 INFO]: MegaLight issued server command: /war
[21:08:08 WARN]: [War] Task #2939 for War v2.0-SNAPSHOT generated an exception
java.lang.NoSuchMethodError: com.sk89q.worldedit.regions.Region.getMinimumPoint()Lcom/sk89q/worldedit/math/BlockVector3;
at com.tommytony.war.utility.Compat.getWorldEditSelection(Compat.java:58) ~[?:?]
at com.tommytony.war.ui.EditOrCreateZoneUI$1$1.run(EditOrCreateZoneUI.java:38) ~[?:?]
at org.bukkit.craftbukkit.v1_13_R2.scheduler.CraftTask.run(CraftTask.java:76) ~[minecraft_server.jar:git-Spigot-2474d93-f0398e4]
at org.bukkit.craftbukkit.v1_13_R2.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:361) [minecraft_server.jar:git-Spigot-2474d93-f0398e4]
at net.minecraft.server.v1_13_R2.MinecraftServer.b(MinecraftServer.java:893) [minecraft_server.jar:git-Spigot-2474d93-f0398e4]
at net.minecraft.server.v1_13_R2.DedicatedServer.b(DedicatedServer.java:417) [minecraft_server.jar:git-Spigot-2474d93-f0398e4]
at net.minecraft.server.v1_13_R2.MinecraftServer.a(MinecraftServer.java:835) [minecraft_server.jar:git-Spigot-2474d93-f0398e4]
at net.minecraft.server.v1_13_R2.MinecraftServer.run(MinecraftServer.java:733) [minecraft_server.jar:git-Spigot-2474d93-f0398e4]
at java.lang.Thread.run(Thread.java:748) [?:1.8.0_181]

IMegaLightI commented:
[21:09:37 INFO]: MegaLight issued server command: /setzone test
[21:09:37 ERROR]: null
org.bukkit.command.CommandException: Unhandled exception executing command 'setzone' in plugin War v2.0-SNAPSHOT
at org.bukkit.command.PluginCommand.execute(PluginCommand.java:46) ~[minecraft_server.jar:git-Spigot-2474d93-f0398e4]
at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:139) ~[minecraft_server.jar:git-Spigot-2474d93-f0398e4]
at org.bukkit.craftbukkit.v1_13_R2.CraftServer.dispatchCommand(CraftServer.java:699) ~[minecraft_server.jar:git-Spigot-2474d93-f0398e4]
at net.minecraft.server.v1_13_R2.PlayerConnection.handleCommand(PlayerConnection.java:1648) ~[minecraft_server.jar:git-Spigot-2474d93-f0398e4]
at net.minecraft.server.v1_13_R2.PlayerConnection.a(PlayerConnection.java:1483) ~[minecraft_server.jar:git-Spigot-2474d93-f0398e4]
at net.minecraft.server.v1_13_R2.PacketPlayInChat.a(PacketPlayInChat.java:45) ~[minecraft_server.jar:git-Spigot-2474d93-f0398e4]
at net.minecraft.server.v1_13_R2.PacketPlayInChat.a(PacketPlayInChat.java:1) ~[minecraft_server.jar:git-Spigot-2474d93-f0398e4]
at net.minecraft.server.v1_13_R2.PlayerConnectionUtils.a(SourceFile:10) ~[minecraft_server.jar:git-Spigot-2474d93-f0398e4]
at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_181]
at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_181]
at net.minecraft.server.v1_13_R2.SystemUtils.a(SourceFile:199) [minecraft_server.jar:git-Spigot-2474d93-f0398e4]
at net.minecraft.server.v1_13_R2.MinecraftServer.b(MinecraftServer.java:900) [minecraft_server.jar:git-Spigot-2474d93-f0398e4]
at net.minecraft.server.v1_13_R2.DedicatedServer.b(DedicatedServer.java:417) [minecraft_server.jar:git-Spigot-2474d93-f0398e4]
at net.minecraft.server.v1_13_R2.MinecraftServer.a(MinecraftServer.java:835) [minecraft_server.jar:git-Spigot-2474d93-f0398e4]
at net.minecraft.server.v1_13_R2.MinecraftServer.run(MinecraftServer.java:733) [minecraft_server.jar:git-Spigot-2474d93-f0398e4]
at java.lang.Thread.run(Thread.java:748) [?:1.8.0_181]
Caused by: java.lang.NoSuchMethodError: com.sk89q.worldedit.regions.Region.getMinimumPoint()Lcom/sk89q/worldedit/math/BlockVector3;
at com.tommytony.war.utility.Compat.getWorldEditSelection(Compat.java:58) ~[?:?]
at com.tommytony.war.command.SetZoneCommand.handle(SetZoneCommand.java:29) ~[?:?]
at com.tommytony.war.command.WarCommandHandler.handle(WarCommandHandler.java:141) ~[?:?]
at com.tommytony.war.War.onCommand(War.java:344) ~[?:?]
at org.bukkit.command.PluginCommand.execute(PluginCommand.java:44) ~[minecraft_server.jar:git-Spigot-2474d93-f0398e4]
... 15 more

cmastudios commented:
It's because of the subtext I added saying it only works for WorldEdit released after Oct. 19. Working on a change for the currently released version.

cmastudios commented:
The following build should work for worldedit version 7.0.0-beta-01: https://s3-us-west-2.amazonaws.com/maven.cmastudios.me/com/tommytony/war/2.0-SNAPSHOT/war-2.0-20181119.201311-3.jar

cmastudios committed:
Dumb WorldEdit unstable API

Now works for 7.0.0-beta-01

IMegaLightI commented:
you are the best man, thank you because I did not find worldedit of October 19th and the plugin works perfectly nice job man. ps: Sorry for my english I am french ;) <3

IMegaLightI commented:
![2018-11-19_21 16 36](https://user-images.githubusercontent.com/6020767/48732825-04103300-ec41-11e8-9385-3012b6ac1683.png)

IMegaLightI commented:
![2018-11-19_21 22 47](https://user-images.githubusercontent.com/6020767/48732933-610be900-ec41-11e8-98b8-eda883bbb47f.png)

cmastudios commented:
D'accord :) dites-moi si vous trouverez d'autres erreurs.

IMegaLightI commented:
wait what your french ? or you learned French

cmastudios commented:
Studied at university. Thanks for testing!

IMegaLightI commented:
ok nice anyway well played because I know that French is a language very difficult to learn

cmastudios committed:
Remove deprecated legacy materials

By deleting the entire unused methods

IMegaLightI commented:
nice

IMegaLightI commented:
if one day you want to see your plugin in work you can come see on our server

play.battlecraft.ovh
